Gems/Multiplayer/Code/Source/NetworkEntity/EntityReplication/EntityReplicator.cpp

```diff
@@ -405,6 +409,62 @@ namespace Multiplayer
         return m_replicationManager.GetResendTimeoutTimeMs();
     }
 
+    bool EntityReplicator::IsReadyToActivate() const
+    {
+        const AZ::Entity* entity = m_entityHandle.GetEntity();
+        AZ_Assert(entity, "Entity replicator entity unexpectedly missing");
+
+        const NetworkHierarchyChildComponent* hierarchyChildComponent = entity->FindComponent<NetworkHierarchyChildComponent>();
+        const NetworkHierarchyRootComponent* hierarchyRootComponent = nullptr;
+
+        if (hierarchyChildComponent == nullptr)
+        {
+            // child and root hierarchy components are mutually exclusive
+            hierarchyRootComponent = entity->FindComponent<NetworkHierarchyRootComponent>();
+        }
+
+        if ((hierarchyChildComponent && hierarchyChildComponent->IsHierarchicalChild())
+            || (hierarchyRootComponent && hierarchyRootComponent->IsHierarchicalChild()))
+        {
+            // if hierarchy is enabled for the entity, check if the parent is available
+            if (const NetworkTransformComponent* networkTransform = entity->FindComponent<NetworkTransformComponent>())
+            {
+                const NetEntityId parentId = networkTransform->GetParentEntityId();
+                /*
+                 * For root entities attached to a level, a network parent won't be set.
+                 * In this case, this entity is the root entity of the hierarchy and it will be activated first.
+                 */
+                if (parentId != InvalidNetEntityId)
+                {
+                    ConstNetworkEntityHandle parentHandle = GetNetworkEntityManager()->GetEntity(parentId);
+
+                    const AZ::Entity* parentEntity = parentHandle.GetEntity();
+                    if (parentEntity && parentEntity->GetState() == AZ::Entity::State::Active)
+                    {
+                        if (bg_debugHierarchyActivation)
+                        {
+                            AZLOG_DEBUG(
+                                "Entity %s asking for activation - granted",
+                                entity->GetName().c_str());
+                        }
+                        return true;
+                    }
+
+                    if (bg_debugHierarchyActivation)
+                    {
+                        AZLOG_DEBUG(
```

Comment:
consider switching stuff like this to a trace log:
AZLOG(mps_DebugHierarchyActivation, ...);
Instead of a log + cvar.

Reply:
What is the perf hit of that?

Reply:
I'll have to look that up. Does that unite the two?

Reply:
nope, it should be much more efficient than the two unless you invoke complicated logic as part of your format arguments (which you should never do anyways)

Reply:
Converted over.